Étape 1 : Comprendre le dataset

Avant de nettoyer, il faut répondre à quelques questions.

In [1]:
from google.colab import files

fichier = files.upload()

Saving world-data-2023.csv to world-data-2023.csv


**1- Importer le Dataset**

In [ ]:
import pandas as pd

df = pd.read_csv("world-data-2023.csv")

df.head()
df.info()
df.describe(include="all")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 35 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Country                                    195 non-null    object 
 1   Density
(P/Km2)                            195 non-null    object 
 2   Abbreviation                               188 non-null    object 
 3   Agricultural Land( %)                      188 non-null    object 
 4   Land Area(Km2)                             194 non-null    object 
 5   Armed Forces size                          171 non-null    object 
 6   Birth Rate                                 189 non-null    float64
 7   Calling Code                               194 non-null    float64
 8   Capital/Major City                         192 non-null    object 
 9   Co2-Emissions                              188 non-null    object 
 10  CPI                       

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,...,Out of pocket health expenditure,Physicians per thousand,Population,Population: Labor force participation (%),Tax revenue (%),Total tax rate,Unemployment rate,Urban_population,Latitude,Longitude
count,195,195,188,188,194,171,189.000000,194.000000,192,188,...,188,188.000000,194,176,169,183,176,190,194.000000,194.000000
unique,195,137,188,168,194,105,NaN,NaN,192,184,...,160,NaN,194,145,119,156,164,190,NaN,NaN
top,Afghanistan,25,AF,17.40%,"652,230","2,000",NaN,NaN,Kabul,"28,284",...,40.50%,NaN,"38,041,754",65.10%,19.50%,36.60%,4.59%,"9,797,273",NaN,NaN
freq,1,7,1,3,1,7,NaN,NaN,1,2,...,3,NaN,1,3,4,4,3,1,NaN,NaN
mean,NaN,NaN,NaN,NaN,NaN,NaN,20.214974,360.546392,NaN,NaN,...,NaN,1.839840,NaN,NaN,NaN,NaN,NaN,NaN,19.092351,20.232434
std,NaN,NaN,NaN,NaN,NaN,NaN,9.945774,323.236419,NaN,NaN,...,NaN,1.684261,NaN,NaN,NaN,NaN,NaN,NaN,23.961779,66.716110
min,NaN,NaN,NaN,NaN,NaN,NaN,5.900000,1.000000,NaN,NaN,...,NaN,0.010000,NaN,NaN,NaN,NaN,NaN,NaN,-40.900557,-175.198242
25%,NaN,NaN,NaN,NaN,NaN,NaN,11.300000,82.500000,NaN,NaN,...,NaN,0.332500,NaN,NaN,NaN,NaN,NaN,NaN,4.544175,-7.941496
50%,NaN,NaN,NaN,NaN,NaN,NaN,17.950000,255.500000,NaN,NaN,...,NaN,1.460000,NaN,NaN,NaN,NaN,NaN,NaN,17.273849,20.972652
75%,NaN,NaN,NaN,NaN,NaN,NaN,28.750000,506.750000,NaN,NaN,...,NaN,2.935000,NaN,NaN,NaN,NaN,NaN,NaN,40.124603,48.281523


**2 - Nombres de lignes et de colonnes**

In [ ]:
df.shape

(195, 35)

Le Dataset contient 195 observations, correspondant chacune à un Pays, et 35 variables décrivant différents indicateurs démographiques, économiques, sanitaires et environnementaux.

**3 - Afficher les premières lignes**

In [ ]:
df.head(10)

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,...,Out of pocket health expenditure,Physicians per thousand,Population,Population: Labor force participation (%),Tax revenue (%),Total tax rate,Unemployment rate,Urban_population,Latitude,Longitude
0,Afghanistan,60,AF,58.10%,"652,230","323,000",32.49,93.0,Kabul,"8,672",...,78.40%,0.28,"38,041,754",48.90%,9.30%,71.40%,11.12%,"9,797,273",33.939110,67.709953
1,Albania,105,AL,43.10%,"28,748","9,000",11.78,355.0,Tirana,"4,536",...,56.90%,1.20,"2,854,191",55.70%,18.60%,36.60%,12.33%,"1,747,593",41.153332,20.168331
2,Algeria,18,DZ,17.40%,"2,381,741","317,000",24.28,213.0,Algiers,"150,006",...,28.10%,1.72,"43,053,054",41.20%,37.20%,66.10%,11.70%,"31,510,100",28.033886,1.659626
3,Andorra,164,AD,40.00%,468,NaN,7.20,376.0,Andorra la Vella,469,...,36.40%,3.33,"77,142",NaN,NaN,NaN,NaN,"67,873",42.506285,1.521801
4,Angola,26,AO,47.50%,"1,246,700","117,000",40.73,244.0,Luanda,"34,693",...,33.40%,0.21,"31,825,295",77.50%,9.20%,49.10%,6.89%,"21,061,025",-11.202692,17.873887
5,Antigua and Barbuda,223,AG,20.50%,443,0,15.33,1.0,"St. John's, Saint John",557,...,24.30%,2.76,"97,118",NaN,16.50%,43.00%,NaN,"23,800",17.060816,-61.796428
6,Argentina,17,AR,54.30%,"2,780,400","105,000",17.02,54.0,Buenos Aires,"201,348",...,17.60%,3.96,"44,938,712",61.30%,10.10%,106.30%,9.79%,"41,339,571",-38.416097,-63.616672
7,Armenia,104,AM,58.90%,"29,743","49,000",13.99,374.0,Yerevan,"5,156",...,81.60%,4.40,"2,957,731",55.60%,20.90%,22.60%,16.99%,"1,869,848",40.069099,45.038189
8,Australia,3,AU,48.20%,"7,741,220","58,000",12.60,61.0,Canberra,"375,908",...,19.60%,3.68,"25,766,605",65.50%,23.00%,47.40%,5.27%,"21,844,756",-25.274398,133.775136
9,Austria,109,AT,32.40%,"83,871","21,000",9.70,43.0,Vienna,"61,448",...,17.90%,5.17,"8,877,067",60.70%,25.40%,51.40%,4.67%,"5,194,416",47.516231,14.550072


Les premières lignes montrent que chaque ligne représente un pays. Les colonnes regroupent des informations comme la population, le PIB, l'espérance de vie, les émissions de CO₂, etc. Certaines colonnes numériques semblent toutefois contenir des caractères spéciaux ($, %, ,) qui devront être nettoyés.

**4. Les Noms des Colonnes**

In [ ]:
list(df.columns)

['Country',
 'Density\n(P/Km2)',
 'Abbreviation',
 'Agricultural Land( %)',
 'Land Area(Km2)',
 'Armed Forces size',
 'Birth Rate',
 'Calling Code',
 'Capital/Major City',
 'Co2-Emissions',
 'CPI',
 'CPI Change (%)',
 'Currency-Code',
 'Fertility Rate',
 'Forested Area (%)',
 'Gasoline Price',
 'GDP',
 'Gross primary education enrollment (%)',
 'Gross tertiary education enrollment (%)',
 'Infant mortality',
 'Largest city',
 'Life expectancy',
 'Maternal mortality ratio',
 'Minimum wage',
 'Official language',
 'Out of pocket health expenditure',
 'Physicians per thousand',
 'Population',
 'Population: Labor force participation (%)',
 'Tax revenue (%)',
 'Total tax rate',
 'Unemployment rate',
 'Urban_population',
 'Latitude',
 'Longitude']

Les noms des colonnes contiennent des espaces et des majuscules. Ils seront uniformisés, (minuscules et underscores) afin de faciliter leur utilisation avec Pandas et SQL.

**5. Les types de données**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 35 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Country                                    195 non-null    object 
 1   Density
(P/Km2)                            195 non-null    object 
 2   Abbreviation                               188 non-null    object 
 3   Agricultural Land( %)                      188 non-null    object 
 4   Land Area(Km2)                             194 non-null    object 
 5   Armed Forces size                          171 non-null    object 
 6   Birth Rate                                 189 non-null    float64
 7   Calling Code                               194 non-null    float64
 8   Capital/Major City                         192 non-null    object 
 9   Co2-Emissions                              188 non-null    object 
 10  CPI                       

Dans le **commentaire** :

Plusieurs colonnes numériques sont reconnues comme des chaînes de caractères (object), probablement à cause de la présence de symboles ($, %, ,). Elles devront être converties en types numériques après nettoyage.

**6. Les statistiques descriptives**

In [ ]:
df.describe()


,Birth Rate,Calling Code,Fertility Rate,Infant mortality,Life expectancy,Maternal mortality ratio,Physicians per thousand,Latitude,Longitude
count,189.000000,194.000000,188.000000,189.000000,187.000000,181.000000,188.000000,194.000000,194.000000
mean,20.214974,360.546392,2.698138,21.332804,72.279679,160.392265,1.839840,19.092351,20.232434
std,9.945774,323.236419,1.282267,19.548058,7.483661,233.502024,1.684261,23.961779,66.716110
min,5.900000,1.000000,0.980000,1.400000,52.800000,2.000000,0.010000,-40.900557,-175.198242
25%,11.300000,82.500000,1.705000,6.000000,67.000000,13.000000,0.332500,4.544175,-7.941496
50%,17.950000,255.500000,2.245000,14.000000,73.200000,53.000000,1.460000,17.273849,20.972652
75%,28.750000,506.750000,3.597500,32.700000,77.500000,186.000000,2.935000,40.124603,48.281523
max,46.080000,1876.000000,6.910000,84.500000,85.400000,1150.000000,8.420000,64.963051,178.065032


Tu obtiens :

moyenne ;
minimum ;
maximum ;
quartiles.

Puis :

In [ ]:
df.describe(include="object")

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Capital/Major City,Co2-Emissions,CPI,CPI Change (%),...,Largest city,Minimum wage,Official language,Out of pocket health expenditure,Population,Population: Labor force participation (%),Tax revenue (%),Total tax rate,Unemployment rate,Urban_population
count,195,195,188,188,194,171,192,188,178,179,...,189,150,190,188,194,176,169,183,176,190
unique,195,137,188,168,194,105,192,184,175,86,...,188,114,76,160,194,145,119,156,164,190
top,Afghanistan,25,AF,17.40%,"652,230","2,000",Kabul,"28,284",106.58,1.80%,...,S����,$2.00,English,40.50%,"38,041,754",65.10%,19.50%,36.60%,4.59%,"9,797,273"
freq,1,7,1,3,1,7,1,2,2,7,...,2,3,31,3,1,3,4,4,3,1


Pour les colonnes texte

**Commentaire :**

Les statistiques descriptives permettent de détecter des valeurs potentiellement aberrantes et de mieux comprendre la distribution des variables numériques.

7. **Les Valeurs Manquantes**

In [ ]:
(df.isnull().sum() / len(df) * 100).sort_values(ascending=False)

,0
Minimum wage,23.076923
Tax revenue (%),13.333333
Armed Forces size,12.307692
Gasoline Price,10.256410
Unemployment rate,9.743590
Population: Labor force participation (%),9.743590
CPI,8.717949
CPI Change (%),8.205128
Currency-Code,7.692308
Maternal mortality ratio,7.179487


Certaines variables présentent des valeurs manquantes, leur traitement sera défini lors de l'étape de nettoyage en fonction de leur importance pour l'analyse.

**8- LES DOUBLONS**

In [ ]:
df.duplicated().sum()

np.int64(0)

**Commentaire : **
Cette vérification permet de s'assurer qu'aucun pays n'est présent plusieurs fois dans le dataset.

Premières Anomalies Observées:
- Plusieurs colonnes numériques sont importées comme des chaines de carctères (object).
- Des valeurs manquantes sont présentes dans certaines variables.
- Les noms de colonnes contiennent des espaces et des majuscules.
- Certaines valeurs numériques incluent des symboles ($, %, ,) qui empèchent leur conversion directe.
